src/Schema/Keywords/OneOf.php

```diff
@@ -55,17 +56,24 @@ public function validate($data, array $oneOf) : void
         // Validate against all schemas
         $matchedCount    = 0;
         $schemaValidator = new SchemaValidator($this->validationDataType);
+        $innerExceptions = [];
+
         foreach ($oneOf as $schema) {
             try {
                 $schemaValidator->validate($data, $schema, $this->dataBreadCrumb);
                 $matchedCount++;
             } catch (SchemaMismatch $e) {
-                // that did not match... its ok
+                $innerExceptions[] = $e;
             }
         }
 
         if ($matchedCount !== 1) {
```

Comment:
For new exception we should distinguish between zero and >1 matches. For zero matches - use aggregated exception. for >1 we should just provide some useful info, since there is nothing bad with unmatched ones

Reply:
thanks for fast reply. what do you thin about this:
- keeping logic like it is for zero matches
- add another exception type "TooManyValidSchemas"* that works like InvalidSchemaCombination but instead of $innerExceptions it stores an array of $validSchemas

* if you have an idea for some better name please tell me.

Reply:
`InvalidSchemaCombination` naming should be changed to `NotEnoughValidSchemas` then or something similar, since current naming covers both cases. All the other parts of the idea sounds good.

Reply:
i see, that makes sense. i've updated the code. feel free to check it again :)